src/btree/bt_handle.c

```diff
@@ -694,7 +694,8 @@ __btree_conf(WT_SESSION_IMPL *session, WT_CKPT *ckpt, bool is_ckpt)
     WT_ASSERT(session, ckpt->write_gen >= ckpt->run_write_gen);
 
     /* If this is the first time opening the tree this run. */
-    if (F_ISSET(session, WT_SESSION_IMPORT) || ckpt->run_write_gen < conn->base_write_gen)
+    if (F_ISSET(session, WT_SESSION_IMPORT) || ckpt->run_write_gen < conn->base_write_gen ||
+      F_ISSET(btree, WT_BTREE_DISAGGREGATED))
```

Comment:
This is still not 100% correct. It should be (F_ISSET(btree, WT_BTREE_DISAGGREGATED) && !disagg->leader). In the leader's case, when we open a btree, there may be txnids a running transaction cannot see.

Reply:
Yes, we should put in something like that. I'll do this for now, but I think we would still have to improve upon it:
* We'll need to ensure that there are no problems on follower step-up.
* We'll also need to ensure that there are no problems when starting the leader without local files. In that case, we would also need to ensure that txn IDs are ignored.

I'll leave SLS-556 open for that investigation...

Reply:
Actually, I will leave the code as in the PR, because our current priority for M3 is to have a leader starting without local files, in which case the leader would need to ignore the transaction IDs as well. I will add a FIXME for this.

Reply:
For the leader restarting case, (ckpt->run_write_gen < conn->base_write_gen) should be true.